### ISSUE TITLE
[Bug] Dovecot trying to create files in users home directory, permission failed

### ISSUE THREAD

**Issue by sandroshu** [hestiacp/hestiacp#3964]
### Describe the bug

Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user>/dovecot.autoexpunge.lock) failed: Permission denied

This error shows up every time a user gets the INBOX folder via IMAP with Roundcube.

Sieve scripts are not working either.

Also Spam goes into both "Spam" folder and "Junk", but I only have "Junk" in configuration. (I haven't messed with the mail config).

The only thing I did previously is adding SnappyMail feature.

### Tell us how to replicate the bug

Enable dovecot logging.
Login to mail account.
Check log.

### Which components are affected by this bug?

Mail (Exim, Dovecot)

### Hestia Control Panel Version

1.8.7

### Operating system

Ubuntu 22.04

### Log capture

```shell
Aug 27 22:21:11 imap(staff@<domain1>.hu)<210941><RmKXUe0D3NV/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:21:11 imap(staff@<domain1>.hu)<210941><RmKXUe0D3NV/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:22:11 imap(staff@<domain1>.hu)<211233><ZvoqVe0DiOx/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:22:11 imap(staff@<domain1>.hu)<211233><ZvoqVe0DiOx/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:22:11 imap(staff@<domain1>.hu)<211233><ZvoqVe0DiOx/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:22:38 imap(staff@<domain1>.hu)<211505><GA/QVu0D1Mp/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:22:38 imap(staff@<domain1>.hu)<211505><GA/QVu0D1Mp/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:22:38 imap(staff@<domain1>.hu)<211505><GA/QVu0D1Mp/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:22:48 imap(staff@<domain1>.hu)<211530><L4heV+0DrKB/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:22:48 imap(staff@<domain1>.hu)<211530><L4heV+0DrKB/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:22:48 imap(staff@<domain1>.hu)<211530><L4heV+0DrKB/AAAB>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user1>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user1>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:23:46 imap(<user2>@<domain2>.hu)<211819><jnTLWu0DIFRU7FeJ>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user2>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user2>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:23:46 imap(<user2>@<domain2>.hu)<211819><jnTLWu0DIFRU7FeJ>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user2>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user2>/dovecot.autoexpunge.lock) failed: Permission denied
Aug 27 22:23:46 imap(<user2>@<domain2>.hu)<211819><jnTLWu0DIFRU7FeJ>: Error: autoexpunge: Couldn't create dovecot.autoexpunge.lock lock: file_create_locked(/home/<user2>/dovecot.autoexpunge.lock) failed: safe_mkstemp(/home/<user2>/dovecot.autoexpunge.lock) failed: Permission denied
```

**Comment by Aartsie** [comment]
Hmmm its missing the username in the path's so that's why there is no permission i think.

**Comment by jaapmarcus** [comment]
I wonder why it makesnin /home/user/ and not in /home/user/mail folder

**Comment by sandroshu** [comment]
I have checked the config in /usr/local/hestia/install/common/dovecot and I see no difference.
I have no idea why is it trying to create that file in the users home instead of under the mailboxes path (/home/user/mail/domain/box/here).

In /etc/exim4/domains/xxx.hu/passwd and /etc/exim4/domains/xxx.hu/accounts file I see the home for all mailboxes are set to /home/user, but I believe that is correct, since in dovecot config it is set correctly to mail/domain/mailbox
Sending/Receiving mail works correctly, only sieve does not work and I think the issue is that it is looking for the script in the wrong path just like the dovecot.autoexpunge.lock file.

**Comment by jaapmarcus** [comment]
https://github.com/hestiacp/hestiacp/blob/04e34fd29bc36a4b59a5216ff05cb4f668119966/install/deb/exim/exim4.conf.4.95.template#L446

This is for Exim for receiving .. With the default settings ( No Sieve)

**Comment by jaapmarcus** [comment]
How did you activate the sieve?

**Comment by sandroshu** [comment]
I have enabled it at install.

I have that line in /etc/exim4/exim4.conf.template

Original install was on Ubuntu 20.04, I have upgraded it to 22.04 a few months ago following this: https://hestiacp.com/docs/server-administration/os-upgrades.html

**Comment by sandroshu** [comment]
I've checked what install_sieve.sh does and strange...
The patches that this script does to the exim4.conf.template were missing. After running the 2 sed commands the exim config is okay now, however I got: "T=dovecot_virtual_delivery defer (0): Child process of dovecot_virtual_delivery transport returned 75 (could mean temporary error) from command: /usr/lib/dovecot/dovecot-lda"
That is because of this: "Fatal: Couldn't load required plugin /usr/lib/dovecot/modules/lib11_imap_quota_plugin.so: dlopen() failed: /usr/lib/dovecot/modules/lib11_imap_quota_plugin.so: undefined symbol: command_unregister"

I fixed this by removing imap_quota from plugins from 15-lda.conf in dovecot config.

Now sieve works as expected.

Sieve did work before however, I have no idea how this happened to the config. Also I don't know why imap_quota was in lda.
